app/services/search/postgres/username.rb

```diff
@@ -0,0 +1,33 @@
+module Search
+  module Postgres
+    class Username
+      ATTRIBUTES = %i[
+        id
+        name
+        profile_image
+        username
+      ].freeze
+
+      def self.search_documents(term)
+        users = search_users(term)
+
+        # We need to re-map profile_image as the profile_image_90 method on the
+        # User model
+        users.map do |user|
+          {
+            id: user.id,
+            name: user.name,
+            profile_image_90: user.profile_image_90,
+            username: user.username
+          }
+        end.as_json
```

Comment:
I feel like this could be improved - any suggestions?

Reply:
Maybe I'm misunderstanding something, but the comment says ` # We need to re-map profile_image as the profile_image_90 method` and the code says `profile_image_90: user.profile_image_90`.

Apart from that:

```ruby
user.as_json(only: [:id, :name, :username])
  .merge("profile_image" => user.profile_image_90) # as_json returns string keys
```

Reply:
Would this work?

```ruby
search_users(term).as_json(only: %i[id name username profile_image_90])
```

Reply:
I was under the impression that `profile_image_90` should be mapped to `profile_image`, at least that's what the comment says, but not what the code does right now. If that assumption is wrong your suggestion is the way to go.

Reply:
Updated this to

```ruby
user.as_json(only: [:id, :name, :username])
  .merge("profile_image" => user.profile_image_90)
```

It is supposed to be `profile_image_90` @citizen428. What I originally was trying to describe about re-mapping was that in attributes we grab `profile_image`:

```ruby
ATTRIBUTES = %i[
  id
  name
  profile_image
  username
]

def self.search_users(term)
  ::User
    .search_by_username(term)
    .select(*ATTRIBUTES)
end
```

And we need `profile_image_90` which is a model method and not an attribute.

@jgaskins I tried
```ruby
search_users(term).as_json(only: %i[id name username profile_image_90])
```

But that doesn't seem to work since `profile_image_90` isn't an attribute.

Reply:
Ugh, ActiveRecord distinguishing attributes from methods is irritating. It was worth a shot, though! 🙂